### DISCUSSION TITLE
Is there any way to access wiki or `Documents & Pages` which is the same level as Work Items?

### DISCUSSION THREAD

**Discussion by JaneSoo** [jesper-raemaekers/python-polarion#115] [Q&A]
I want to access the wiki page and do some export from my Python script but I could not find something in the documentation page that specifies on how to achieve that. Is the feature not available or do I miss something while reading?
Here is the URL: https://polarion.engineering.abc.com/polarion/#/project/project_name/wiki/  which is the same level as the work items URL: https://polarion.engineering.abc.com/polarion/#/project/project_name/workitems/. 

I linked my question in Stackoverflow here for ref: https://stackoverflow.com/questions/76278961/python-polarion-api-how-to-access-wiki-documentations-and-pages

**Comment by jesper-raemaekers** [comment]
Hi JaneSoo,

This is not supported by this library at the moment. but you can easily implement this using this example below:

```
# open a project
pol = polarion.Polarion('', '', '')
prj = pol.getProject('Python')

# get the service
service = pol.getService('Tracker')
# get all wiki pages in the Design space. not needed if you have the uri for the wiki page
wiki_pages = service.getWikiPages(prj.id, 'Design')
uri = wiki_pages[0].uri # here is only one wiki page, so get that uri
print(uri)

wiki_page = service.getWikiPageByUri(uri)
print(wiki_page.homePageContent)

```
Example output:
```
subterra:data-service:objects:/default/Python${WikiPage}{moduleFolder}Design#wiki_test
{
    'type': 'text/plain',
    'content': 'THIS PAGE HAS NO CONTENT YET.\n\nYOU CAN CREATE AND SAVE CONTENT NOW USING THIS EDITOR',
    'contentLossy': False
}
```

Let me know if this works for you and please also link the stackoverflow post to this discussion as it could help people in the future.

**Reply by JaneSoo** [reply]
Thank you so much, this is very useful. However, when I tried to do `wiki_page = service.getWikiPageByUri(uri)` and encountered this exception `*** zeep.exceptions.Fault: Not authorized.`

**Reply by JaneSoo** [reply]
Never mind, when I started again. It worked. I guess I lost the session or something. Thank you @jesper-raemaekers !

**Reply by JaneSoo** [reply]
I am sorry to bother you again. But not all the things under the space are listed, for example, the document type `info page` is not there.  I tried to dig into your source code to find something similar to `getWikiPages` but I could not find anything and not even the function `getWikiPages` itself.  I am I looking into the wrong place? I checked out to the `master` branch.

**Comment by jesper-raemaekers** [comment]
All functions which are accessed by the service.<something> calls are dynamically constructed. A reference of those can be found on <your polarion url>/polarion/ws/services.
I've checked that list and these seems to be no way to retrieve the info pages.

**Reply by JaneSoo** [reply]
After all the searching around, it seems that the doc I want to fetch is a `live document`. I tried this `document = project.getDocument(f'spaces[0]/{document.name}')` but there is an error as below:

```
*** AttributeError: 'Document' object has no attribute 'title'
```

**Reply by jesper-raemaekers** [reply]
Seems like a live document cannot be fetched like a normal document. I cannot find any way to achieve this currently.
